src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1935,18 +1936,33 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
         auto gcc_polymorphic = typecheck_gcc_polymorphic_builtin(
           identifier, expr.arguments(), f_op.source_location()))
       {
-        if(!symbol_table.has_symbol(gcc_polymorphic->get_identifier()))
+        irep_idt identifier_w_type = gcc_polymorphic->get_identifier();
```

Comment:
Make some tests for the name mangling? Are there any particular cases where you might foresee the mangling producing an invalid name?

Reply:
As those names are only for internal use we are rather free to use whatever strings are formed here. But certainly I should at least have a test exercising this, which I have now added.